menpo/shape/mesh/base.py

```diff
@@ -236,6 +309,257 @@ def n_tris(self):
+        Parameters
+        ----------
+        target_mesh :   `TriMesh`
+            A TriMesh whose points are used to find the differences(subtrahend)
+        scalar_range : `tuple'
+            The scalar range of  the colorbar, default=(0,2)
+        scale_value : `int'
+            The scale value of the differences, to go to mm, default : 100
+        type_cmap : `cmap'
+            Type of the colormap, default : 'hot', it can be:
+            'Accent','Blues','BrBG','BuGn','BuPu','CMRmap','Dark2', 'GnBu',
+            'Greens','Greys', 'OrRd', 'Oranges', 'PRGn', 'Paired', 'Pastel1',
+            'Pastel2', 'PiYG', 'PuBu', 'PuBuGn', 'PuOr', 'PuRd', 'Purples',
+            'RdBu', 'RdGy', 'RdPu', 'RdYlBu', 'RdYlGn', 'Reds', 'Set1',
+            'Set2', 'Set3', 'Spectral', 'Vega10', 'Vega20', 'Vega20b',
+            'Vega20c', 'Wistia', 'YlGn', 'YlGnBu', 'YlOrBr', 'YlOrRd',
+            'afmhot', 'autumn', 'binary', 'black-white', 'blue-red',
+            'bone', 'brg', 'bwr','cool' or 'coolwarm' or 'copper',
+            'cubehelix', 'file', 'flag', 'gist_earth', 'gist_gray',
+            'gist_heat', 'gist_ncar', 'gist_rainbow', 'gist_stern',
+            'gist_yarg', 'gnuplot', 'gnuplot2', 'gray', 'hot', 'hsv',
+            'inferno', 'jet', 'magma', 'nipy_spectral', 'ocean', 'pink',
+            'plasma', 'prism', 'rainbow', 'seismic', 'spectral' 'spring',
+            'summer', 'terrain', 'viridis', 'winter'
+        show_statistics : `bool'
+            If statistics like mean, standard deviation and max error will be
+            shown in the window,
+            default:False
+        automatic_id : `bool`
+            If ``True'', an automatic text is created for figure_id
+        inline : 'bool', False
+               If True, the viewer will be in the Jupyter cell using K3dwidgets
+               If False, the viewer will open a new window using Mayavi
+
+        Returns
+        -------
+        v : `Scene`
+            Handle to  mayavi scene or a K3dwidgetsHeatmapViewer3d object
+        scaled_distances_between_meshes : `np.array'
+            An array with the scaled distances between the
+            correspoding vertices.
+        Raises
+        ------
+        ValueError
+        """
+        source_mesh = self
+        source_n_vertices = source_mesh.points.shape[0]
+        target_mesh_n_vertices = target_mesh.points.shape[0]
+
+        if not source_n_vertices == target_mesh_n_vertices:
+            first_part_string = 'Source mesh has {} vertices while target mesh has {}'.format(source_n_vertices,
+                                                                                              target_mesh_n_vertices)
+            print(first_part_string)
+            subject = source_mesh.points
+            template = target_mesh
+            X = template.points
+
+            tree = cKDTree(X)
+            dist, indx = tree.query(subject, k=1)
+
+            target_mesh = TriMesh(X[indx], source_mesh.trilist)
```

Comment:
This is a copy?

Reply:
I suppose yes. Perhaps a not so clever way to save some lines of code. I have to check this again.